cmd/manager/main.go

```diff
@@ -146,11 +149,20 @@ func main() {
 		statusupdater.NewModuleStatusUpdater(client),
 		operatorNamespace,
 	)
-
 	if err = mc.SetupWithManager(mgr, constants.KernelLabel); err != nil {
 		cmd.FatalError(setupLogger, err, "unable to create controller", "name", controllers.ModuleReconcilerName)
 	}
 
+	mnc := controllers.NewModuleNMCReconciler(
+		client,
+		kernelAPI,
+		nmcHandler,
```

Comment:
Should this one be renamed to `nmcAPI` to be consistent with the rest of the API passed to it (or to the Module reconciler)?

Reply:
This is a controller, we are not creating/returning an Interface ( API), but the struct itself, as the rest of the reconcilers/controllers, that's why there is no API in the name

Reply:
`nmcHandler` is a public interface. Why don't we call it `nmcAPI` instead?

Reply:
moved the NMC-Handler API into the controller, since it implements controller logic